src/transaction/mvcc_table.cpp

```diff
@@ -707,17 +707,35 @@ mvcctable::get_global_oldest_visible () const
 }
 
 MVCCID
-mvcctable::update_global_oldest_visible ()
+mvcctable::update_global_oldest_visible (const MVCCID pts_oldest_visible /* = MVCCID_LAST */)
```

Comment:
Some fixes:
> "skip vacuum regardless of MVCCID values because the cluster is in an inconsistent state with at least one PTS booting-up (or being shut down)"

not during shutdown

> scalability: update for vacuum to execute or not (!= MVCCID_ALL_VISIBLE and != MVCCID_LAST)

It is updated even if ==MVCCID_LAST, at this point, just the mvccid of ATS is considered and the vaccum can be run on it. I think you might've said the same one, but rephrase it with my words to be clear.

I'll add a comment on the overloaded one.